### PR TITLE
Middleware: Simplifications

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Chore PR to simplify backend middleware code a bit, in preparation for a deeper PR to remove dependency on global state.

### CODE REVIEW

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -25,14 +25,25 @@ func RegisterServiceWithPriority(instance Service, priority Priority) {
 }
 
 func RegisterService(instance Service) {
-	services = append(services, &Descriptor{
+	Register(&Descriptor{
```
Comment (wbrowne) [inline pkg/registry/registry.go:28]: Just wanted to confirm you meant to exclude the `RegisterServiceWithPriority` func from using `Register`?

**Review by wbrowne — APPROVED** [review]
LGTM! Just a small remark

File: pkg/registry/registry.go (hunk 1 above)
Comment (aknuds1) [reply pkg/registry/registry.go:28]: @wbrowne I will check.

Comment (aknuds1) [reply pkg/registry/registry.go:28]: @wbrowne Looking at the code, I simply didn't think of it

Comment (aknuds1) [reply pkg/registry/registry.go:28]: Thanks, made the change!

**Pull request merged**